modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaSearchVocabulariesMethodCheck.java

```diff
@@ -34,45 +38,63 @@ protected String doProcess(
 		boolean failedValidation = false;
 		boolean replaced = false;
 
-		Matcher matcher = _searchVocabulariesPattern.matcher(content);
-
-		while (matcher.find()) {
-			String methodCall = JavaSourceUtil.getMethodCall(
-				content, matcher.start());
-
-			if (!hasClassOrVariableName(
-					"AssetVocabularyService", content, content, methodCall) &&
-				!hasClassOrVariableName(
-					"AssetVocabularyLocalService", content, content,
-					methodCall)) {
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
 
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
 				continue;
 			}
 
-			List<String> parameterList = JavaSourceUtil.getParameterList(
-				methodCall);
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
 
-			if (parameterList.size() != 5) {
-				continue;
-			}
+			String javaMethodContent = javaMethod.getContent();
 
-			String[] parameterTypes = {"long", "long", "String", "int", "int"};
+			Matcher matcher = _searchVocabulariesPattern.matcher(
+				javaMethodContent);
 
-			if (!hasParameterTypes(
-					content, content, ArrayUtil.toStringArray(parameterList),
-					parameterTypes)) {
+			while (matcher.find()) {
+				String methodCall = JavaSourceUtil.getMethodCall(
+					javaMethodContent, matcher.start());
 
-				failedValidation = true;
+				if (!hasClassOrVariableName(
+						"AssetVocabularyService", content, content,
+						methodCall) &&
+					!hasClassOrVariableName(
+						"AssetVocabularyLocalService", content, content,
+						methodCall)) {
 
-				continue;
-			}
+					continue;
+				}
+
+				List<String> parameterList = JavaSourceUtil.getParameterList(
+					methodCall);
 
-			String newMethod = _addOrReplaceParameters(
-				matcher.group(0), methodCall, parameterList);
+				if (parameterList.size() != 5) {
+					continue;
+				}
 
-			newContent = StringUtil.replace(newContent, methodCall, newMethod);
+				String[] parameterTypes = {
+					"long", "long", "String", "int", "int"
+				};
 
-			replaced = true;
+				if (!hasParameterTypes(
+						javaMethodContent, javaMethodContent,
+						ArrayUtil.toStringArray(parameterList),
+						parameterTypes)) {
+
+					failedValidation = true;
```

Comment:
@nicolas-sss, @TamyrisTorres, do we really need this variable?

Why not do something like the following?
```
if (!hasParameterTypes(
	javaMethodContent, javaMethodContent,
	ArrayUtil.toStringArray(parameterList), parameterTypes)) {

	continue;
}

addMessage(
	fileName,
	StringBundler.concat(
		"Could not resolve types of searchVocabularies method. ",
		"The method signature has changed to searchVocabularies(",
		"long companyId, long[] groupIds, String title, int[] ",
		"visibilityTypes, int start, int end, Sort sort). Fill ",
		"the new parameters manually."));
```